tests/components/caldav/test_calendar.py

```diff
@@ -914,9 +932,18 @@ async def test_get_events_custom_calendars(hass, calendar):
     assert await async_setup_component(hass, "calendar", {"calendar": config})
     await hass.async_block_till_done()
 
-    entity = hass.data["calendar"].get_entity("calendar.private_private")
-    events = await entity.async_get_events(
-        hass, datetime.date(2015, 11, 27), datetime.date(2015, 11, 28)
+    events = await get_api_events(
+        "calendar.private_private",
+        "2015-11-27",
+        "2015-11-28",
     )
-    assert len(events) == 1
-    assert events[0]["summary"] == "This is a normal event"
+    assert events == [
+        {
+            "description": "Surprisingly rainy",
+            "end": "2017-11-27T10:00:00-08:00",
```

Comment:
The start and end time here doesn't match the start and end date of the parameters to `get_api_events`. Is something wrong?

Reply:
Good catch, I didn't notice this. The issue here is the calendar mock ignores the start/end calls and always just returns the events -- since it's not a fake. I've kept that behavior but fixed the readability to make it more clear what's happening.